components/ui/Modal/Modal.tsx

```diff
@@ -7,45 +7,52 @@ import {
   enableBodyScroll,
   clearAllBodyScrollLocks,
 } from 'body-scroll-lock'
-
+import FocusTrap from '@lib/focus-trap'
 interface Props {
   className?: string
   children?: any
   open?: boolean
   onClose: () => void
+  onEnter?: () => void | null
 }
 
-const Modal: FC<Props> = ({ children, open, onClose }) => {
+const Modal: FC<Props> = ({ children, open, onClose, onEnter = null }) => {
   const ref = useRef() as React.MutableRefObject<HTMLDivElement>
 
+  const handleKey = (e: KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      return onClose()
+    }
+  }
```

Comment:
Use `useCallback` here and then add `handleKey` to the deps of the `useEffect` below, that way the function `onClose` will be called properly if it's changed by the parent component for any reason.

Reply:
I read somewhere that it actually there was not point in wrapping this into `useCallback` for performance reasons. Only, in case of multiple instantiations that resulted in the same functions. Will add it just in case.

Reply:
Is not for performance in this case though, is to avoid using an old version of `onClose`, which is very hard to debug. We could have a linter to be more restrictive with that.